lib/record_store/zone/config.rb

```diff
@@ -4,15 +4,18 @@ class Config
       include ActiveModel::Validations
 
       attr_reader :ignore_patterns, :provider, :supports_alias
-      alias supports_alias? supports_alias
 
       validate :validate_zone_config
 
       def initialize(ignore_patterns: [], provider: nil, supports_alias: nil)
         @ignore_patterns = ignore_patterns
         @provider = provider
         @supports_alias = supports_alias
+      end
+
+      def supports_alias?
         @supports_alias = Provider.const_get(provider).supports_alias? if @supports_alias.nil? && valid_provider?
```

Comment:
Since this is an reader, it shouldn't be modifying any internal state. How about `@supports_alias || (valid_provider? && Provider.const_get(provider).supports_alias?)`?